src/languages/go.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  createPatternMatchers,
+  argumentMatcher,
+  leadingMatcher,
+  conditionMatcher,
+  trailingMatcher,
+  childMatcher,
+  cascadingMatcher,
+  patternMatcher,
+} from "../util/nodeMatchers";
+import {
+  NodeMatcherAlternative,
+  ScopeType,
+  SelectionWithEditor,
+} from "../typings/Types";
+import { getNodeRange } from "../util/nodeSelectors";
+import { SyntaxNode } from "web-tree-sitter";
+
+// Generated by the following command:
+// `curl https://raw.githubusercontent.com/tree-sitter/tree-sitter-go/master/src/node-types.json | jq '[.[] | select(.type == "_statement" or .type == "_simple_statement") | .subtypes[].type]'`
+const STATEMENT_TYPES = [
+  "_expression",
+  "assignment_statement",
+  "dec_statement",
+  "inc_statement",
+  "send_statement",
+  "short_var_declaration",
+  "_simple_statement",
+  "block",
+  "break_statement",
+  "const_declaration",
+  "continue_statement",
+  "defer_statement",
+  "empty_statement",
+  "expression_switch_statement",
+  "fallthrough_statement",
+  "for_statement",
+  "go_statement",
+  "goto_statement",
+  "if_statement",
+  "labeled_statement",
+  "return_statement",
+  "select_statement",
+  "type_declaration",
+  "type_switch_statement",
+  "var_declaration"
+];
+
+const nodeMatchers: Partial<Record<ScopeType, NodeMatcherAlternative>> = {
+  map: "composite_literal",
+  list: ["composite_literal", "slice_type", "array_type"],
+  statement: STATEMENT_TYPES,
+  string: ["interpreted_string_literal", "raw_string_literal"],
+  ifStatement: "if_statement",
+  functionCall: "call_expression",
+  comment: "comment",
+  namedFunction: ["function_declaration", "method_declaration"],
+  type: cascadingMatcher(
+    patternMatcher("pointer_type"),
+    patternMatcher("qualified_type"),
+    patternMatcher("type_identifier"),
+    patternMatcher("function_declaration[result]"),
+    patternMatcher("method_declaration[result]"),
+  ),
+  functionName: ["function_declaration[name]", "method_declaration[name]"],
+  anonymousFunction: "func_literal",
+  condition: conditionMatcher("*[condition]"),
+  argumentOrParameter: argumentMatcher("argument_list", "parameter_list"),
+  collectionItem: ["keyed_element.*", "element.*"],
```

Comment:
How come you don't want
```suggestion
  collectionItem: ["keyed_element", "element"],
```

Reply:
:+1: